lib/spack/spack/compilers/clang.py

```diff
@@ -206,6 +206,7 @@ def extract_version_from_output(cls, output):
             r'^Apple LLVM version ([^ )]+)|'
             # Normal clang compiler versions are left as-is
             r'clang version ([^ )]+)-svn[~.\w\d-]*|'
+            r'clang version ([^ )]+)-[~.\w\d-]*|'
```

Comment:
What's the point of the trailing pipe?

Reply:
From the docs:

> A|B, where A and B can be arbitrary REs, creates a regular expression that will match either A or B. An arbitrary number of REs can be separated by the '|' in this way. This can be used inside groups (see below) as well. As the target string is scanned, REs separated by '|' are tried from left to right. When one pattern completely matches, that branch is accepted. This means that once A matches, B will not be tested further, even if it would produce a longer overall match. In other words, the '|' operator is never greedy. To match a literal '|', use \|, or enclose it inside a character class, as in [|].

It's a way to test for alternative patterns using a single regular expression.

Reply:
Ooooh, I assumed this was a list, not a single multi-line string. Thanks!